src/rest-server/src/models/vc.js

```diff
@@ -80,6 +71,254 @@ class VirtualCluster {
+            if (err) {
+              return callback(err);
+            } else {
+              return callback(null);
+            }
+          });
+        }
+      }
+    });
+  }
+
+  stopVc(vcName, callback) {
+    this.getVcList((vcList, err) => {
+      if (err) {
+        return callback(err);
+      } else if (!vcList) {
+        // Unreachable
+        logger.warn('list virtual clusters error, no virtual cluster found');
+      } else {
+        if (!vcList.hasOwnProperty(vcName)) {
+          return callback(createError('Not Found', 'NoVirtualClusterError', `Vc ${vcName} not found, can't stop`));
+        } else {
+          let data = {'pendingStop': {}};
+          data['pendingStop'][vcName] = null;
+
+          logger.debug('raw data to generate: ', data);
+          const vcdataXml = this.generateUpdateInfo(data);
+          logger.debug('Xml send to yarn: ', vcdataXml);
+          this.sendUpdateInfo(vcdataXml, (err) => {
+            if (err) {
+              return callback(err);
+            } else {
+              return callback(null);
+            }
+          });
+        }
+      }
+    });
+  }
+
+  activeVc(vcName, callback) {
+    this.getVcList((vcList, err) => {
+      logger.debug(err);
+      if (err) {
+        return callback(err);
+      } else if (!vcList) {
+        // Unreachable
+        logger.warn('list virtual clusters error, no virtual cluster found');
+      } else {
+        if (!vcList.hasOwnProperty(vcName)) {
+          return callback(createError('Not Found', 'NoVirtualClusterError', `Vc ${vcName} not found, can't active`));
+        } else {
+          let data = {'pendingActive': {}};
+          data['pendingActive'][vcName] = null;
+
+          logger.debug('raw data to generate: ', data);
+          const vcdataXml = this.generateUpdateInfo(data);
+          logger.debug('Xml send to yarn: ', vcdataXml);
+          this.sendUpdateInfo(vcdataXml, (err) => {
+            if (err) {
+              return callback(err);
+            } else {
+              return callback(null);
+            }
+          });
+        }
+      }
+    });
+  }
+
+  removeVc(vcName, callback) {
+    this.getVcList((vcList, err) => {
+      if (err) {
+        return callback(err);
+      } else if (!vcList) {
+        // Unreachable
+        logger.warn('list virtual clusters error, no virtual cluster found');
+      } else {
+        if (!vcList.hasOwnProperty('default')) {
+          return callback(createError('Not Found', 'NoVirtualClusterError', `No default vc found, can't free quota`));
+        } else if (!vcList.hasOwnProperty(vcName)) {
+          return callback(createError('Not Found', 'NoVirtualClusterError', `Can't delete a nonexistent vc ${vcName}`));
+        } else {
+          this.stopVc(vcName, (err) => {
+            if (err) {
+              return callback(err);
+            } else {
+              let defaultQuotaIfUpdated = vcList['default']['capacity'] + vcList[vcName]['capacity'];
```

Comment:
Could you make sure about these 2 values will be Number? `+` will be treated as string concat when one of these is string.

Reply:
Yes, this two value comes from YARN, they are both number.